finality-aleph/src/import.rs

```diff
@@ -64,16 +64,28 @@ where
         number: NumberFor<Block>,
         justification: Justification,
     ) -> Result<(), SendJustificationError<Block>> {
-        debug!(target: "afa", "Importing justification for block #{:?}", number);
+        debug!(target: "afa", "Importing justification for block {:?}", number);
         if justification.0 != ALEPH_ENGINE_ID {
             return Err(SendJustificationError::Consensus(Box::new(
                 ConsensusError::ClientImport("Aleph can import only Aleph justifications.".into()),
             )));
         }
         let justification_raw = justification.1;
 
-        let aleph_justification = AlephJustification::decode(&mut &*justification_raw)
-            .map_err(|_| SendJustificationError::Decode)?;
+        let aleph_justification: AlephJustification = {
+            if let Ok(justification) =
+                AlephJustification::decode(&mut &*(justification_raw.clone()))
+            {
+                justification
+            } else if let Ok(justification) = AlephJustificationV1::decode(&mut &*justification_raw)
+            {
+                debug!(target: "afa", "Justification for block {:?} decoded correctly as V1", number);
+                justification.into()
+            } else {
+                warn!(target: "afa", "Justification for block {:?} decoded incorrectly", number);
```

Comment:
Shouldn't this log line happen at the caller, since here we are returning an error anyway?

Reply:
This was slightly rewritten, but I think it now addresses this comment.